test/meanshift.cpp

```diff
@@ -65,38 +65,29 @@ void meanshiftTest(string pTestFile)
 
     for (size_t testId=0; testId<testCount; ++testId) {
 
-        af_array inArray        = 0;
-        af_array inArray_f32    = 0;
-        af_array outArray       = 0;
-        af_array goldArray      = 0;
-        af_array goldArray_f32  = 0;
-        dim_t nElems            = 0;
+        inFiles[testId].insert(0,string(TEST_DIR "/meanshift/"));
+        outFiles[testId].insert(0,string(TEST_DIR "/meanshift/"));
 
-        inFiles[testId].insert(0,string(TEST_DIR"/meanshift/"));
-        outFiles[testId].insert(0,string(TEST_DIR"/meanshift/"));
+        try {
+            af_dtype type = (af_dtype)af::dtype_traits<T>::af_type;
 
-        ASSERT_EQ(AF_SUCCESS, af_load_image(&inArray_f32, inFiles[testId].c_str(), isColor));
-        ASSERT_EQ(AF_SUCCESS, conv_image<T>(&inArray, inArray_f32));
+            array input = loadImage(inFiles[testId].c_str(), isColor).as(type);
```

Comment:
Tests should use the C API whenever possible. The C++ API should be tested separately.

Reply:
I can refactor this to use C API.